### ISSUE TITLE
Reproducibility of Uncertainty Experiment

### ISSUE THREAD

**Issue by sporsho** [wjmaddox/swa_gaussian#8]
Hi, 
I was trying to reproduce the uncertainty experiment. I used the same command used in the README file of this repo. For training SWAG, I ran `run_swag.py --data_path=./data --epochs=300 --dataset=CIFAR10 --save_freq=300  --model=PreResNet164 --lr_init=0.1 --wd=3e-4 --swa --swa_start=161 --swa_lr=0.01 --cov_mat --use_test
      --dir=./snap`

Afterwards, I ran `uncertainty.py  --data_path=./data --dataset=CIFAR10 --model=PreResNet164 --use_test
      --cov_mat --method=SWAG --scale=0.5 --file=<checkpoint_file> --save_path=<path_to_save_var>` 

Test accuracy aligns with the result from paper but NLL value sky-rockets to 1000+ where it is around 0.12 in the paper. Can you check if I have missed something in the command line params, also please tell if I need to do some other tuning to reproduce the NLL values.

**Comment by wjmaddox** [comment]
Sorry for not responding to this issue earlier. 

What version of Pytorch are you using?

I also believe that the NLL number you are seeing could be the sum NLL --- so that dividing it by 10000 should give you an answer along the lines of 0.12.

**Comment by sporsho** [comment]
Hi, thanks for your kind reply. I checked your code again. It is printing the sum of NLL (in uncertainty.py). Kindly update the code with the average NLL so other people will not get confused. 

Cheers !!
